dvc/tree/gdrive.py

```diff
@@ -510,35 +512,88 @@ def exists(self, path_info, use_dvcignore=True):
         else:
             return True
 
-    def _list_paths(self, prefix=None):
-        if not self._ids_cache["ids"]:
-            return
+    def _gdrive_list_ids(self, query_ids):
+        query = " or ".join(
+            f"'{query_id}' in parents" for query_id in query_ids
+        )
+        query = f"({query}) and trashed=false"
+        return self._gdrive_list(query)
+
+    def _ls_recursive(self, path_info, detail=False):
+        root_path = path_info.path
+        seen_paths = set()
+
+        dir_ids = [self._ids_cache["ids"].copy()]
+        while dir_ids:
+            query_ids = {
+                dir_id: dir_name
+                for dir_id, dir_name in dir_ids.pop().items()
+                if posixpath.commonpath([root_path, dir_name])
+                if dir_name not in seen_paths
```

Comment:
probably should be by `dir_id` also? since GD allows multiple dirs with the same path

Reply:
Makes sense, yes!